crates/autopilot/src/run_loop.rs

```diff
@@ -445,8 +445,17 @@ impl RunLoop {
         .await
         {
             futures::future::Either::Left((res, _)) => res,
-            futures::future::Either::Right((res, _)) => {
-                res.map_err(SettleError::Failure).map(|tx| tx.tx_hash)
+            futures::future::Either::Right((driver_result, onchain_task)) => {
+                let driver_tx_hash = driver_result.map_err(SettleError::Failure)?.tx_hash;
+                let onchain_tx_hash = onchain_task.await?;
+                if onchain_tx_hash != driver_tx_hash {
```

Comment:
Completely overlooked this in the review of the original PR but my initial idea was to only allow solvers to report when they gave up submitting a tx on-chain and always take the tx hash from the autopilot. This PR would essentially accomplish the same with the additional overhead of making the solver report back the tx hash that will ultimately get ignored.

Reply:
Removed `/settle` response body.